### ISSUE TITLE
Image is not zooming while continuous clicking

### ISSUE THREAD

**Issue by karthickramasamy08** [spinningarrow/zoom-vanilla.js#47]
Added the zooming files on my blog page. It is working for normal cases, but i am trying to click the image multiple time continuously it is not respond(can't zooming). Can you please check and help me to resolve this problem.

**Comment by spinningarrow** [comment]
Can you elaborate on the problem a little? Does it happen on all browsers? Could you send me a video showing the issue?

**Comment by jetibest** [comment]
Yes, it happens on any browser, no need for a video. You can easily reproduce it by reducing your internet speed and disable cache in responsive dev mode of browser. And click again on the image to zoom, while the overlay has not been shown yet, because the image for the overlay is still loading.

The problem is that when the `zoom(event)` is called, it creates the `activeZoom` object together with the event-listeners to remove the `activeZoom` object again. However, the overlay is added when the `img.onload` is called.

So if you already destroyed `activeZoom` while `img.onload` has not called yet, then the overlay will show up, but without an `activeZoom` object. This results in the major bug the overlay cannot be removed.

There are several solutions to this problem possible:

 - You may add `if(!document.querySelector('.zoom-overlay')) return;` to the `closeActiveZoom(options)` function. So that the `activeZoom` cannot be destroyed when the overlay is not there yet.

 - Another solution would involve aborting the `img.onload` handler. Make the `img` globally available in `vanillaZoom`, and then set `img.onload = null` in the `close()` and `dispose()` functions. I chose this one, as it is the easiest to implement, and most robust.

 - Another solution is to already add the overlay immediately, and also showing a loading-icon. Of course this only applies if the image is not cached, and therefore cannot be shown immediately (for instance, any image that has been dynamically generated by the server and is not supposed to be cached) - otherwise the few milliseconds the loading-icon is shown is not worth the trouble.

**Comment by spinningarrow** [comment]
@jetibest thank you for the reproduction steps! I see the issue now as well. I think option 1 sounds quite straightforward - are there any downsides to that?

Also thanks for the suggestion about fullWidth/naturalWidth. I'll look into it. So far I've avoided making changes to the core of the library that I forked from, but it might be time to revisit that!

**Comment by jetibest** [comment]
Of course, using naturalWidth and naturalHeight is more elegant.

I cannot think of any special downsides to the first option. The only thing to consider is that `.zoom-overlay` is a pretty generic class name, and there might be collisions. But that potential problem is not unique to this specific issue.

**Comment by kirillgroshkov** [comment]
I just noticed this issue in my environment as well (more often in dev mode, using nuxt.js).

Would be nice to have fixed soon.
